client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryWithProtostreamAnnotationsTest.java

```diff
@@ -0,0 +1,266 @@
+import static org.junit.Assert.*;
+
+/**
+ * Tests for remote queries over HotRod on a local cache using RAM directory.
+ *
+ * @author Adrian Nistor
+ */
+@org.testng.annotations.Test(testName = "client.hotrod.query.RemoteQueryWithProtostreamAnnotationsTest", groups = "functional")
+public class RemoteQueryWithProtostreamAnnotationsTest extends SingleCacheManagerTest {
+
+   private static final String TEST_CACHE_NAME = "testCache";
+
+   private HotRodServer hotRodServer;
+   private RemoteCacheManager remoteCacheManager;
+   private RemoteCache<Integer, Memo> remoteCache;
+
+   @ProtoDoc("@Indexed")
+   @ProtoMessage(name = "Memo")
+   public static class Memo {
+
+      private int id;
+
+      private String text;
+
+      private Author author;
+
+      @ProtoDoc("@IndexedField(index = false, store=false)")
+      @ProtoField(number = 10, required = true)
+      public int getId() {
+         return id;
+      }
+
+      public void setId(int id) {
+         this.id = id;
+      }
+
+      @ProtoDoc("@IndexedField")
+      @ProtoField(number = 20)
+      public String getText() {
+         return text;
+      }
+
+      public void setText(String text) {
+         this.text = text;
+      }
+
+      @ProtoDoc("@IndexedField")
+      @ProtoField(number = 30)
+      public Author getAuthor() {
+         return author;
+      }
+
+      public void setAuthor(Author author) {
+         this.author = author;
+      }
+
+      @Override
+      public String toString() {
+         return "Memo{id=" + id + ", text='" + text + '\'' + ", author=" + author + '}';
+      }
+   }
+
+   public static class Author {
+
+      private int id;
+
+      private String name;
+
+      public int getId() {
+         return id;
+      }
+
+      public void setId(int id) {
+         this.id = id;
+      }
+
+      public String getName() {
+         return name;
+      }
+
+      public void setName(String name) {
+         this.name = name;
+      }
+
+      @Override
+      public String toString() {
+         return "Author{id=" + id + ", name='" + name + "'}";
+      }
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      org.infinispan.configuration.cache.ConfigurationBuilder builder = new org.infinispan.configuration.cache.ConfigurationBuilder();
+      builder.dataContainer()
+            .keyEquivalence(ByteArrayEquivalence.INSTANCE)
+            .valueEquivalence(ByteArrayEquivalence.INSTANCE)
+            .indexing().index(Index.ALL)
+            .addProperty("default.directory_provider", "ram")
+            .addProperty("lucene_version", "LUCENE_CURRENT");
+
+      cacheManager = TestCacheManagerFactory.createCacheManager(builder);
+      cacheManager.defineConfiguration(TEST_CACHE_NAME, builder.build());
+      cache = cacheManager.getCache(TEST_CACHE_NAME);
+
+      hotRodServer = HotRodClientTestingUtil.startHotRodServer(cacheManager);
```

Comment:
In spite of using custom cache manager configuration, and custom remote cache client configuration, I'd rather see this class extend `org.infinispan.client.hotrod.test.SingleHotRodServerTest` than carry on duplicating code...